lasso/emme.py

```diff
@@ -268,145 +268,190 @@ def create_emme_network(
+                    (links_df.tollbooth != 0)
+                ) 
+            )
+        ) | links_df["has_bus_on_link"] | links_df["managed_lane_connector"] | (links_df["cntype"] == "MAZ") # Maz is filtered out by links_df.fg <= maximum_ft 
     ].copy()
 
-    model_tables["link_table"] = drive_links_df.to_dict('records')
-    model_tables["node_table"] = drive_nodes_df.to_dict('records')
-
-    return model_tables
-
-def prepare_table_for_maz_drive_network(
-    nodes_df,
-    links_df,
-    parameters,
-):
-
-    """
-    prepare model table for maz-scale drive network, in which there are no centroids, drop taz nodes and connectors
-    keep links that are drive_access == 1 and assignable == 1
-
-    Arguments:
-        nodes_df -- node database
-        links_df -- link database
-    
-    Return:
-        dictionary of model network settings
-    """
+    if not regenerate_connectors:
+        model_tables["connector_table"] = links_df[
+            (links_df.A.isin(parameters.taz_N_list)) | 
+            (links_df.B.isin(parameters.taz_N_list))
+        ].to_dict('records')
+    else:
+        # check ranch is installed
+        raise NotImplementedError("Currently Waiting On task Order 6 to finish implementation")
+        if ranch is None:
+            raise ImportError("package 'ranch' is not installed, please go to https://github.com/BayAreaMetro/Ranch for install instructions")
+        if shapes_df is None:
+            raise ValueError("shapes_df must be provided to regenerate connectors")
+        if taz_zone_shapefile_path is None:
+            raise ValueError("taz_zone_shapefile_path must be passed in when regenerate_connectors=True")
+        if maz_zone_shapefile_path is None:
+            raise ValueError("maz_zone_shapefile_path must be passed in when regenerate_connectors=True")
+
+        # regenerate connectors
+        ranch_dir = find_parent_ranch_dir(ranch.__file__)
+        ranch_params=ranch.Parameters(
+            ranch_base_dir=ranch_dir,
+        )
+        
+        links_df = links_df[links_df["cntype"] != "MAZ"]
+        ranch_roadway = ranch.Roadway(nodes_df, links_df.drop(columns=["index_left", "index_right"], errors="ignore"), shapes_df, ranch_params)
+        buid_taz_node_id
+        ranch_roadway.build_centroid_connectors(build_taz_active_modes=True, build_maz_drive=True, 
+            input_taz_polygon_file=taz_zone_shapefile_path, 
+            input_maz_polygon_file=maz_zone_shapefile_path, 
+        )
 
-    model_tables = dict()
 
-    # no centroids, drop taz nodes and connectors
+        model_tables["connector_table"] = ranch_roadway.links_df[
+            (ranch_roadway.links_df.A.isin(parameters.taz_N_list)) | 
+            (ranch_roadway.links_df.B.isin(parameters.taz_N_list))
+        ].to_dict('records')
 
-    model_tables["centroid_table"] = []
+        # add the new maz links back in 
+        drive_links_df = pd.concat(
+            [
+                drive_links_df, 
```

Comment:
I believe this includes the old maz connectors.

Reply:
less concerned about this, it is behind a not implemented error block.

Updated line 425 so that drive links df is drops maz.

![image](https://github.com/user-attachments/assets/29128a1b-446b-4eb9-9cb1-8ea16ba37527)